staging/src/k8s.io/apiserver/pkg/cel/library/cost.go

```diff
@@ -55,6 +55,18 @@ type CostEstimator struct {
 	SizeEstimator checker.CostEstimator
 }
 
+const (
+	// shortest repeatable selector requirement that allocates a values slice is 2 characters: k,
+	shortestRequirement = float64(2)
+	costPerRequirement  = common.ListCreateBaseCost + common.StructCreateBaseCost
+)
+
+func selectorCostEstimate(selectorLength float64) uint64 {
```

Comment:
@jpbetz can you sanity check this method of estimating cost for a selector based on string length?

Reply:
LGTM.

The `parseCost` and `maxRequirements` math looks sane and safe.  The object construction costs were originally intended to reflect the cost of constructing data literals within the CEL runtime. Using them here as a proxy measure for golang allocs should certainly bound the cost.